egs/tedlium/s5_r2/local/run_segmentation_long_utts.sh

```diff
@@ -0,0 +1,264 @@
+# cleaned_b -
+# "--max-junk-proportion=0.5 --max-deleted-words-kept-when-merging=10"
+# cleaned_c -
+# "--max-junk-proportion=0.2 --max-deleted-words-kept-when-merging=6 --min-split-point-duration=0.3"
+
+# %WER 18.7 | 507 17783 | 84.0 11.0 5.0 2.8 18.7 91.7 | -0.119 | exp/tri3_reseg_a_cleaned/decode_nosp_dev_rescore/score_15_0.0/ctm.filt.filt.sys
+# %WER 18.6 | 507 17783 | 84.0 11.0 4.9 2.7 18.6 91.5 | -0.092 | exp/tri3_reseg_a_cleaned_b/decode_nosp_dev_rescore/score_15_0.0/ctm.filt.filt.sys
+# %WER 18.6 | 507 17783 | 84.1 10.8 5.0 2.7 18.6 92.1 | -0.114 | exp/tri3_reseg_a_cleaned_c/decode_nosp_dev_rescore/score_15_0.0/ctm.filt.filt.sys
+
+# %WER 17.7 | 1155 27500 | 84.5 11.4 4.0 2.2 17.7 86.8 | -0.020 | exp/tri3_reseg_a_cleaned/decode_nosp_test_rescore/score_14_0.0/ctm.filt.filt.sys
+# %WER 17.3 | 1155 27500 | 84.8 11.2 4.1 2.1 17.3 86.8 | -0.002 | exp/tri3_reseg_a_cleaned_b/decode_nosp_test_rescore/score_15_0.0/ctm.filt.filt.sys
+# %WER 17.7 | 1155 27500 | 84.6 11.4 4.1 2.3 17.7 86.6 | -0.018 | exp/tri3_reseg_a_cleaned_c/decode_nosp_test_rescore/score_14_0.0/ctm.filt.filt.sys
+
+## Use silence and pronunciation probs estimated from resegmented data
+# %WER 18.2 | 507 17783 | 84.6 10.8 4.5 2.9 18.2 92.5 | -0.037 | exp/tri3_reseg_a/decode_a_dev_rescore/score_16_0.0/ctm.filt.filt.sys
+# %WER 16.9 | 1155 27500 | 85.5 11.0 3.5 2.4 16.9 86.1 | -0.024 | exp/tri3_reseg_a/decode_a_test_rescore/score_14_0.0/ctm.filt.filt.sys
+
+## Use silence and pronunciation probs estimated from resegmented and cleaned up data
+# %WER 18.2 | 507 17783 | 84.4 10.8 4.9 2.6 18.2 92.5 | -0.074 | exp/tri3_reseg_a_cleaned_b/decode_a_cleaned_b_dev_rescore/score_15_0.5/ctm.filt.filt.sys
+# %WER 16.8 | 1155 27500 | 85.4 10.7 3.9 2.1 16.8 86.8 | -0.046 | exp/tri3_reseg_a_cleaned_b/decode_a_cleaned_b_test_rescore/score_14_0.5/ctm.filt.filt.sys
+
+. ./cmd.sh
+. ./path.sh
+
+set -e -o pipefail -u
+
+segment_stage=-9
+cleanup_stage=-1
+cleanup_affix=cleaned_b
+affix=_a
+
+decode_nj=8   # note: should not be >38 which is the number of speakers in the dev set
+              # after applying --seconds-per-spk-max 180.  We decode with 4 threads, so
+              # this will be too many jobs if you're using run.pl.
+
+###############################################################################
+## Simulate unsegmented data directory.
+###############################################################################
+utils/data/convert_data_dir_to_whole.sh data/train data/train_long
+
+utils/subset_data_dir.sh --speakers data/train 2000 data/train_2k
+utils/subset_data_dir.sh --shortest data/train_2k 500 data/train_2k_short500
+
+steps/make_mfcc.sh --cmd "$train_cmd" --nj 32 \
+  data/train_long exp/make_mfcc/train_long mfcc || exit 1
+steps/compute_cmvn_stats.sh data/train_long \
+  exp/make_mfcc/train_long mfcc
+
+steps/train_mono.sh --nj 20 --cmd "$train_cmd" \
+  data/train_2k_short500 data/lang_nosp exp/mono_a
+
+steps/align_si.sh --nj 20 --cmd "$train_cmd" \
```

Comment:
you might want to mention that we're training a system on a small subset of already-segmented data.